src/main/java/org/folio/consortia/controller/SelfController.java

```diff
@@ -0,0 +1,54 @@
+package org.folio.consortia.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.consortia.domain.dto.Payload;
+import org.folio.consortia.domain.dto.UserTenantCollection;
+import org.folio.consortia.rest.resource.SelfApi;
+import org.folio.consortia.service.UserTenantService;
+import org.folio.consortia.utils.TokenUtils;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/consortia/{consortiumId}")
+@RequiredArgsConstructor
+public class SelfController implements SelfApi {
+
+  @Autowired
+  private final UserTenantService userTenantService;
+
+  @Autowired
+  private final FolioExecutionContext folioExecutionContext;
+
+  @Override
+  public ResponseEntity<UserTenantCollection> getSelfUserTenants(UUID consortiumId) {
+    String token = folioExecutionContext
+      .getOkapiHeaders()
+      .get(XOkapiHeaders.TOKEN)
+      .stream()
+      .findFirst()
+      .orElse("");
+
+    if (token == "") {
+      throw new IllegalArgumentException("token is required");
```

Comment:
throw 401 and maybe handle in another way instead of if

Reply:
fixed

Reply:
@NikitaSedyx :- 
1:-removed userId check , because we are already checking in userTenantService.
2:-upgraded to StringUtils.isEmpty() as it will check both for null and empty string so no need to add 2 checks in if.